File: module-server/src/main/java/de/twomartens/wahlrecht/mapper/v1/SeatResultMapper.java
package de.twomartens.wahlrecht.mapper.v1;

import de.twomartens.wahlrecht.model.internal.SeatResult;
import org.mapstruct.CollectionMappingStrategy;
import org.mapstruct.Mapper;
import org.mapstruct.NullValueCheckStrategy;
import org.mapstruct.ReportingPolicy;

@Mapper(collectionMappingStrategy = CollectionMappingStrategy.ADDER_PREFERRED,
    nullValueCheckStrategy = NullValueCheckStrategy.ALWAYS,
    unmappedTargetPolicy = ReportingPolicy.IGNORE)
public interface SeatResultMapper {

  de.twomartens.wahlrecht.model.dto.v1.SeatResult mapToExternal(SeatResult result);

  SeatResult mapToInternal(de.twomartens.wahlrecht.model.dto.v1.SeatResult result);
}

File: module-server/src/main/kotlin/de/twomartens/wahlrecht/mapper/v1/SeatResultMapper.kt
package de.twomartens.wahlrecht.mapper.v1

import de.twomartens.wahlrecht.model.internal.SeatResult
import org.mapstruct.CollectionMappingStrategy
import org.mapstruct.Mapper
import org.mapstruct.NullValueCheckStrategy
import org.mapstruct.ReportingPolicy

@Mapper(
    collectionMappingStrategy = CollectionMappingStrategy.ADDER_PREFERRED,
    nullValueCheckStrategy = NullValueCheckStrategy.ALWAYS,
    unmappedTargetPolicy = ReportingPolicy.IGNORE
)
interface SeatResultMapper {
    fun mapToExternal(result: SeatResult): de.twomartens.wahlrecht.model.dto.v1.SeatResult

    fun mapToInternal(result: de.twomartens.wahlrecht.model.dto.v1.SeatResult): SeatResult
}